test/phonon/ewald.jl

```diff
@@ -0,0 +1,90 @@
+using Test
+using DFTK
+using LinearAlgebra
+using ForwardDiff
+
+include("../testcases.jl")
+
+function ph_compute_reference_ewald(model_supercell)
+    n_atoms = length(model_supercell.positions)
+    n_dim = model_supercell.n_dim
+    T = eltype(silicon.lattice)
+    dynmat_ad = zeros(T, 3, n_atoms, 3, n_atoms)
+    for τ in 1:n_atoms
+        for γ in 1:n_dim
+            displacement = zero.(model_supercell.positions)
+            displacement[τ] = setindex(displacement[τ], one(T), γ)
+            dynmat_ad[:, :, γ, τ] = -ForwardDiff.derivative(zero(T)) do ε
+                cell_disp = (; model_supercell.lattice, model_supercell.atoms,
+                             positions=ε*displacement .+ model_supercell.positions)
+                model_disp = Model(convert(Model{eltype(ε)}, model_supercell); cell_disp...)
+                forces = DFTK.energy_forces_ewald(model_disp; compute_forces=true).forces
+                hcat(Array.(forces)...)
+            end
+        end
+    end
+    hessian_ad = dynmat_to_cart((; model=model_supercell), dynmat_ad)
+    sort(compute_ω²(hessian_ad))
+end
+
+@testset "Dynamical matrix for Ewald" begin
+    tol = 1e-9
+    model = Model(silicon.lattice, silicon.atoms, silicon.positions)
+
+    ω_uc = []
+    for q in phonon.qpoints
+        hessian = dynmat_to_cart((; model), DFTK.dynmat_ewald(model; q))
+        push!(ω_uc, compute_ω²(hessian))
+    end
+    ω_uc = sort!(collect(Iterators.flatten(ω_uc)))
+
+    if REFERENCE_PH_CALC
```

Comment:
This is convoluted and going to lead to problems down the road. Better make this two `@testset`s where one needs a special argument in the testargs to be activated (maybe reuse the `:FAST_TESTS` thing, see e.g. `silicon_pbe`).